src/libraries/Common/src/System/Buffers/ArrayBufferWriter.cs

```diff
@@ -178,10 +180,19 @@ private void CheckAndResizeBuffer(int sizeHint)
 
                 if ((uint)newSize > int.MaxValue)
                 {
-                    newSize = currentLength + sizeHint;
+                    // Attempt to grow by the larger of the minimum size and half of the available size.
+                    growBy = Math.Max(sizeHint, (int.MaxValue - currentLength) / 2 + 1);
```

Comment:
The +1 is to attempt to allocate an extra byte to have it align on a nice boundary

Reply:
How does adding 1 align on a boundary? Can't `currentLength` be wide variety of even or odd numbers?